ibis/backends/pyspark/compiler.py

```diff
@@ -104,6 +104,23 @@ def compile_sql_query_result(t, expr, scope, timecontext, **kwargs):
     return client._session.sql(query)
 
 
+def _can_be_replaced_by_column_name(column_expr, table):
+    """
+    Return whether the given column_expr can be replaced by its literal
+    name, which is True when column_expr and table[column_expr.get_name()]
+    is semantically the same.
+    """
+    # Each check below is necessary to distinguish a pure projection from
+    # other valid selections, such as a mutation that assigns a new column
+    # or changes the value of an existing column.
+    return (
+        isinstance(column_expr.op(), ops.TableColumn)
+        and column_expr.op().table == table
+        and column_expr.get_name() in table.schema()
```

Comment:
Is this check correct? It feels like the fact that the column_expr's name exists in the table.schema didn't say much. When does this check help us?

Reply:
e.g. we want to rule out 
`table.mutate(new_col=table['A'])`